code/modules/mob/living/simple_animal/distortion/_distortion.dm

```diff
@@ -170,22 +189,84 @@
 
 	PostUnmanifest(egoist)
 	PollGhosts(egoist)
+	egoist.forceMove(get_turf(src))
+	qdel(src)
 	return
 
 /mob/living/simple_animal/hostile/distortion/proc/PollGhosts(mob/living/carbon/human/egoist)
+	if(!new_egoist || mind) //A player unmanifested somehow
+		egoist.key = key
+		mind.transfer_to(egoist)
+		return
 	var/list/mob/dead/observer/candidates = pollCandidatesForMob("Do you want to play as [egoist.real_name]?", ROLE_PAI, null, FALSE, 100, egoist)
 	if(LAZYLEN(candidates))
 		var/mob/dead/observer/C = pick(candidates)
 		egoist.key = C.key
 		if(C.mind)
 			C.mind.transfer_to(egoist)
 			egoist.mind.assigned_role = "Civilian"
-	qdel(src)
-	return
+			return
 
 /mob/living/simple_animal/hostile/distortion/proc/PostUnmanifest(mob/living/carbon/human/egoist)
 	return
 
+//Turn into a distortion
+/mob/proc/BecomeDistortion(mob/living/simple_animal/hostile/distortion/chosenmob = null, instant = FALSE, forced = FALSE)
+	var/mob/themob = null
+	var/list/message_list = list(
```

Comment:
items in a multi-line lists should be elevated one indentation up, so instead of
```
var/list/a_list = list(
item,
item2,
)
```
it should be
```
var/list/a_list = list(
	item,
	item2,
)
```